### PR TITLE
[KM181] Manage the lifecycle of RDS Postgres

### PR DESCRIPTION
**Pull request by paulbes**
Manage the lifecycle of RDS Postgres

## Description
This is the first of a few PRs, with this PR we create the cloud formation template for an [RDS Postgres instance](https://aws.amazon.com/rds/postgresql/). In the following ones, we will setup the server and clientside integrations and add the setup into the declarative cluster.yaml file.

The cloud formation stack contains functionality for:

- Enabled automatic storage scaling
- Setting up a generated secrets manager secret with a rotation policy for the admin password
- Creating a security group that can be associated with a pod for gaining access to the RDS Postgres instance
- Creating the RDS Postgres instance itself

## Motivation and Context
The goal of okctl is to provide value to the end-users, we do this by simplifying the application lifecycle, which includes managing commonly used services such as Postgres

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I have updated the release notes (for the next release).

### CODE REVIEW

**Review by deifyed — APPROVED** [review]
Looks both clean and good to me

File: pkg/cfn/components/rotationschedule/rotationschedule.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	return e.StoredName
+}
+
+// Ref returns a cloud formation intrinsic ref to the resource
+func (e *RotationSchedule) Ref() string {
+	return cloudformation.Ref(e.Name())
+}
+
+const (
+	rotateAfterDays = 30
```
Comment (ivaruf) [inline pkg/cfn/components/rotationschedule/rotationschedule.go:41]: Nice :+1:

**Review by ivaruf — APPROVED** [review]

**Review by yngvark — APPROVED** [review]
Looks good!

File: pkg/cfn/components/dbinstance/dbinstance.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+		AllowMajorVersionUpgrade:    false,
+		AutoMinorVersionUpgrade:     true,
+		BackupRetentionPeriod:       backupRetentionDays, // Enables automatic backup
+		CopyTagsToSnapshot:          true,
+		DBInstanceClass:             "db.t3.small", // Should this be configurable?
+		DBName:                      i.DBName,
+		DBParameterGroupName:        i.DBParameterGroup.Ref(),
+		DBSubnetGroupName:           i.DBSubnetGroupName,
+		DeleteAutomatedBackups:      true,
+		DeletionProtection:          false,
```
Comment (yngvark) [inline pkg/cfn/components/dbinstance/dbinstance.go:68]: Any reason for not having this to true? Perhaps an issue for okctl delete cluster?

**Comment by yngvark** [comment]
For the future, consider hardening with some of the stuff mentioned here: https://www.datavail.com/blog/10-best-practices-to-secure-postgresql-aws-rds-aurora/
* IAM authentication
* TLS
* Auditing
* Delete protection
* Random/Custom port
* ... more?

File: pkg/cfn/components/dbinstance/dbinstance.go (hunk 2 above)
Comment (paulbes) [reply pkg/cfn/components/dbinstance/dbinstance.go:68]: I am not sure, and that is probably the problem. It is something we should probably do more research on.

File: pkg/cfn/components/dbinstance/dbinstance.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+			i.DBParameterGroup.Name(),
+			i.MonitoringRole.Name(),
+		},
+	}
+}
+
+// New creates a database instance
+// - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-rds-database-instance.html
+//
+// Things we should consider doing:
```
[the remark this answers is not in the document]
Comment (paulbes) [reply pkg/cfn/components/dbinstance/dbinstance.go:104]: I think we are okay with this for now, but it is definitely something we should consider helping the user with. We probably just need to make the certificates available via a ConfigMap or similar. The problem is that I don't think we can force the user to encrypt the communication.

[the remark this answers is not in the document]
Comment (paulbes) [reply pkg/cfn/components/dbinstance/dbinstance.go:104]: Just checked, we can enforce this! Lets get back to this and your list!

**Pull request merged**